src/Raven.Server/Documents/Indexes/Static/Spatial/SpatialDistanceFieldComparatorSource.cs

```diff
@@ -101,12 +145,66 @@ private double CalculateDistance(int doc, IState state)
                 var pt = shape as Point;
                 if (pt == null)
                     pt = shape.GetCenter();
-                return _spatialField.GetContext().GetDistCalc().Distance(pt, _originPt);
+                double dist = _isGeo ? 
+                    HaverstineDistance(pt.GetY(), pt.GetX(), _originPt.GetY(), _originPt.GetX()) : 
+                    CartesianDistance(pt.GetY(), pt.GetX(), _originPt.GetY(), _originPt.GetX());
+
+                switch (Units)
+                {
+                    case SpatialUnits.Kilometers:
+                        dist *= DistanceUtils.MILES_TO_KM;
+                        break;
+                    case SpatialUnits.Miles:
+                    default:
+                        break;
+                }
+
+                return dist;
+            }
+
+            public static double HaverstineDistance(double lat1, double lng1, double lat2, double lng2)
+            {
+                // from : https://www.geodatasource.com/developers/javascript
+                if ((lat1 == lat2) && (lng1 == lng2))
+                {
+                    return 0;
+                }
+                else
+                {
+                    var radlat1 = DistanceUtils.DEGREES_TO_RADIANS * lat1;
+                    var radlat2 = DistanceUtils.DEGREES_TO_RADIANS * lat2;
+                    var theta = lng1 - lng2;
+                    var radtheta = DistanceUtils.DEGREES_TO_RADIANS * theta;
+                    var dist = Math.Sin(radlat1) * Math.Sin(radlat2) + Math.Cos(radlat1) * Math.Cos(radlat2) 
+                        * Math.Cos(radtheta);
+                    if (dist > 1)
+                    {
+                        dist = 1;
+                    }
+                    dist = Math.Acos(dist);
+                    dist = dist * DistanceUtils.RADIANS_TO_DEGREES;
+                    dist = dist * 60 * 1.1515;
+                    return dist;
+                }
+            }
+
+            public static double CartesianDistance(double lat1, double lng1, double lat2, double lng2)
+            {
+                double result = 0;
+
+                double v = lat1 - lat2;
+                result += (v * v);
+
+                v = lng2 - lng2;
+                result += (v * v);
+
+                return result;
```

Comment:
Based on the above call of this method I assume it should return value in miles? Doesn't it mean we need to convert the longitude and latitude differences calculated here into miles?

Reply:
This is for Cartesian, so no fixed measurement, whatever that values says it is.
This is useful for spatial on flat locations (such as places in warehouse), no sense is miles/km, the unit is whatever the user say it is

Reply:
My comment didn't make sense. I just noticed we do this:
https://github.com/ravendb/ravendb/pull/9367/files#diff-e5a57e10a19c3b4fc6617d363b0188a1R154

only for Haverstine distance, not for Cartesian.
https://github.com/ravendb/ravendb/pull/9367/files#diff-e5a57e10a19c3b4fc6617d363b0188a1R150

Reply:
Right. Same as commented above:

My comment didn't make sense. I just noticed we do this:
https://github.com/ravendb/ravendb/pull/9367/files#diff-e5a57e10a19c3b4fc6617d363b0188a1R154

only for Haverstine distance, not for Cartesian.
https://github.com/ravendb/ravendb/pull/9367/files#diff-e5a57e10a19c3b4fc6617d363b0188a1R150

Reply:
You were correct at the time, I changed that based on your comment